hw/snitch_cluster/src/snitch_cluster_peripheral/snitch_cluster_peripheral_reg.rdl

```diff
@@ -0,0 +1,242 @@
+                    _This is a DMA-local signal_";
+        };
+        DMA_BUF_R_STALL = 15 {
+            desc = "Incremented whenever r_valid = 1 but r_ready = 0.
+                    _This is a DMA-local signal_";
+        };
+        DMA_AW_DONE = 16 {
+            desc = "Incremented whenever AW handshake occurs.
+                    _This is a DMA-local signal_";
+        };
+        DMA_AW_BW = 17 {
+            desc = "Whenever AW handshake occurs, the counter is incremented
+                    by the number of bytes transfered for this transaction
+                    _This is a DMA-local signal_";
+        };
+        DMA_AR_DONE = 18 {
+            desc = "Incremented whenever AR handshake occurs.
+                    _This is a DMA-local signal_";
+        };
+        DMA_AR_BW = 19 {
+            desc = "Whenever AR handshake occurs, the counter is incremented
+                    by the number of bytes transfered for this transaction
+                    _This is a DMA-local signal_";
+        };
+        DMA_R_DONE = 20 {
+            desc = "Incremented whenever R handshake occurs.
+                    _This is a DMA-local signal_";
+        };
+        DMA_R_BW = 21 {
+            desc = "Whenever R handshake occurs, the counter is incremented
+                    by the number of bytes transfered in this cycle
+                    _This is a DMA-local signal_";
+        };
+        DMA_W_DONE = 22 {
+            desc = "Incremented whenvever W handshake occurs.
+                    _This is a DMA-local signal_";
+        };
+        DMA_W_BW = 23 {
+            desc = "Whenever W handshake occurs, the counter is incremented
+                    by the number of bytes transfered in this cycle
+                    _This is a DMA-local signal_";
+        };
+        DMA_B_DONE = 24 {
+            desc = "Incremented whenever B handshake occurs.
+                    _This is a DMA-local signal_";
+        };
+        DMA_BUSY = 25 {
+            desc = "Incremented whenever DMA is busy.
+                    _This is a DMA-local signal_";
+        };
+        ICACHE_MISS = 26 {
+            desc = "Incremented for instruction cache misses.
+                    _This is a hart-local signal_";
+        };
+        ICACHE_HIT = 27 {
+            desc = "Incremented for instruction cache hits.
+                    _This is a hart-local signal_";
+        };
+        ICACHE_PREFETCH = 28 {
+            desc = "Incremented for instruction cache prefetches.
+                    _This is a hart-local signal_";
+        };
+        ICACHE_DOUBLE_HIT = 29 {
+            desc = "Incremented for instruction cache double hit.
+                    _This is a hart-local signal_";
+        };
+        ICACHE_STALL = 30 {
+            desc = "Incremented for instruction cache stalls.
+                    _This is a hart-local signal_";
+        };
+    };
+
+    reg PERF_CNT_SEL {
```

Comment:
Any chance we can define reset values for array regs individually? Then we don't even have to define `PERF_CNT_SEL` as external registers. The idea is that the performance counters will start tracking a couple of different metrics at start up.

Reply:
I don't think there is a straightforward way to specify this. Currently, these registers are specified as external, so the reset values should be applied externally, i.e., here: https://github.com/pulp-platform/snitch_cluster/blob/main/hw/snitch_cluster/src/snitch_cluster_peripheral/snitch_cluster_peripheral.sv#L202-L209

Reply:
The reason why they were defined as externally was because reset values cannot be defined individually with the regtool🤓